arviz/plots/ppplot.py

```diff
@@ -0,0 +1,169 @@
+        InferenceData object containing the posterior/prior data.
+    figsize : tuple
+        Figure size. If None it will be defined automatically.
+    textsize: float
+        Text size scaling factor for labels, titles and lines. If None it will be
+        autoscaled based on figsize.
+    var_names : list
+        List of variables to be plotted.
+    coords : dict
+        Dictionary mapping dimensions to selected coordinates to be plotted.
+        Dimensions without a mapping specified will include all coordinates for
+        that dimension.
+    transform : callable
+        Function to transform data (defaults to None i.e. the identity function)
+    data_pairs : dict
+        Dictionary containing relations between prior data and posterior data.
+        Dictionary structure:
+
+        - key = prior data var_name
+        - value = posterior var_name
+
+        For example, `data_pairs = {'y' : 'y_hat'}`
+        If None, it will assume that the prior data and the posterior
+        data have the same variable name.
+    legend : bool
+        Add legend to figure. By default True.
+    ax: axes, optional
+        Matplotlib axes: a numpy 2d array of matplotlib axes.
+    prior_kwargs : dicts, optional
+        Additional keywords passed to `arviz.plot_kde` for prior group.
+    posterior_kwargs : dicts, optional
+        Additional keywords passed to `arviz.plot_kde` for posterior group.
+    backend: str, optional
+        Select plotting backend {"matplotlib","bokeh"}. Default "matplotlib".
+    backend_kwargs: bool, optional
+        These are kwargs specific to the backend being used. For additional documentation
+        check the plotting method of the backend.
+    show : bool, optional
+        Call backend show function.
+
+    Returns
+    -------
+    axes : matplotlib axes or bokeh figures
+
+    Examples
+    --------
+    Plot the prior/posterior plot for specified vars and coords.
+
+    .. plot::
+        :context: close-figs
+
+        >>> import arviz as az
+        >>> data = az.load_arviz_data('radon')
+        >>> az.plot_pp(data, var_names=["defs"], coords={"team" : ["Italy"]})
+
+    """
+
+    groups = ["prior", "posterior"]
+
+    if transform is not None:
+        data = transform(data)
+
+    if coords is None:
+        coords = {}
+
+    if prior_kwargs is None:
+        prior_kwargs = {}
+
+    if posterior_kwargs is None:
+        posterior_kwargs = {}
+
+    if backend_kwargs is None:
+        backend_kwargs = {}
+
+    if data_pairs is None:
+        data_pairs = {}
+
+    datasets = [getattr(data, group) for group in groups]
+
+    if var_names is None:
+        var_names = list(datasets[0].data_vars)
+    var_names = _var_names(var_names, datasets[0])
+    vars = [var_names]
+    for i in range(1, len(datasets)):
+        var_name =[data_pairs.get(var, var) for var in var_names]
+        var_name = _var_names(var_name,  datasets[i])
+        vars.append(var_name)
```

Comment:
Code inspired in `plot_density` and `plot_forest` while trying to save the distances: here is the [first relevant piece](https://github.com/arviz-devs/arviz/blob/master/arviz/plots/densityplot.py#L159-L162) and here is the [second one](https://github.com/arviz-devs/arviz/blob/master/arviz/plots/forestplot.py#L155).

The trickiest part is var_names handling, while making sure that vars are kept paired (or kept in trios if it were to be extended to prior predictive/observed data/posterior predictive). The code on var names looks good, my only comment is in prevision of adding the 3 plot case, in which case, data_pairs would not work. Maybe it would be better to use a list/list of lists? something like:
* if var_names = None, use as var names all data_vars in first dataset
* if var_name is str, make `var_names = [var_names]
* if var_name is list and var_names[0] is str, var_names = [var_names for _ in datasets]
* otherwise go on and see if it works, var_names should be a list of lists. Note that I used ifs, not elif

Once `datasets` is created, you should be able to use the same code as in `plot_density` and `plot_forest`. First transform, then select coords. Note that `get_coords` should be able to handle a list of datasets and that therefore `coords` can be either a dict or a list of dicts (unlike the description in `plot_forest` docstring specifies), it would look similar to the code below.